megamek/src/megamek/client/ui/swing/boardview/PlanetaryConditionsOverlay.java

```diff
@@ -0,0 +1,405 @@
+        // Extract a color code from the start of the string
+        // used to display headlines if it's there
+        if (s.startsWith("#") && s.length() > 7) {
+            try {
+                int red = Integer.parseInt(s.substring(1, 3), 16);
+                int grn = Integer.parseInt(s.substring(3, 5), 16);
+                int blu = Integer.parseInt(s.substring(5, 7), 16);
+                textColor = new Color(red, grn, blu);
+            } catch (Exception e) {
+                LogManager.getLogger().error("", e);
+            }
+            s = s.substring(7);
+        }
+
+        if (s.length() > 0) {
+            AttributedString text = new AttributedString(s);
+            text.addAttribute(TextAttribute.FONT, new Font(FONT.getFontName(), Font.PLAIN, (int) (FONT.getSize() * GUIPreferences.getInstance().getGUIScale())), 0, s.length());
+
+            graph.setColor(SHADOW_COLOR);
+            graph.drawString(text.getIterator(), x + 1, y + 1);
+            graph.setColor(textColor);
+            graph.drawString(text.getIterator(), x, y);
+        }
+    }
+    
+    /** 
+     * Activates or deactivates the overlay, fading it in or out.
+     * Also saves the visibility to the GUIPreferences so 
+     * MegaMek remembers it. 
+     * */
+    public void setVisible(boolean vis) {
+        visible = vis;
+        GUIPreferences.getInstance().setValue(GUIPreferences.SHOW_PLANETARYCONDITIONS_OVERLAY, vis);
+        if (vis) {
+            fadingIn = true;
+            fadingOut = false;
+        } else {
+            fadingIn = false;
+            fadingOut = true;
+        }
+    }
+
+    public boolean isVisible() {
+        return visible;
+    }
+    
+    @Override
+    public boolean isSliding() {
+        return fadingOut || fadingIn;
+    }
+    
+    @Override
+    public boolean slide() {
+        if (fadingIn) {
+            alpha += FADE_SPEED;
+            if (alpha > 1) {
+                alpha = 1;
+                fadingIn = false;
+            }
+            return true;
+        } else if (fadingOut) {
+            alpha -= FADE_SPEED;
+            if (alpha < 0) {
+                alpha = 0;
+                fadingOut = false;
+            }
+            return true;
+        }
+        return false;
+    }
+    
+    /** Detects phase and turn changes to display Planetary Conditions. */
+    private GameListener gameListener = new GameListenerAdapter() {
+        @Override
+        public void gamePhaseChange(GamePhaseChangeEvent e) {
+            currentPhase = e.getNewPhase();
+            changed = true;
+        }
+        
+        @Override
+        public void gameTurnChange(GameTurnChangeEvent e) {
+            // The active player has changed
+            changed = true;
+        }
+    };
+
+    @Override
+    public void preferenceChange(PreferenceChangeEvent e) {
+        if (e.getName().equals(KeyBindParser.KEYBINDS_CHANGED)) {

```

Comment:
I believe it's not hurtful to just set changed to true here without doing any checks (PreferenceChangeEvents don't happen all the time); this way it'll react to changes to its advanced settings.